util/src/snappy.rs

```diff
@@ -150,22 +150,22 @@ pub fn validate_compressed_buffer(input: &[u8]) -> bool {
 }
 
 const CITA_COMPRESS_SIZE: usize = 40 * 1024;
-pub fn cita_compress(input: Vec<u8>) -> Vec<u8> {
+pub fn cita_compress(input: &[u8]) -> Vec<u8> {
     if input.len() > CITA_COMPRESS_SIZE {
-        compress(&input)
+        compress(input)
     } else {
-        input
+        input.to_vec()
```

Comment:
这样实际上是做了一次clone吧？反而不如直接move进来好？

Reply:
I wrote this part for the old logic (the original bytes has another use after being compressed / decompressed).
I will **remove** this part before **reopen** this PR.

The result of those two functions `cita_compress` and `cita_decompress` is not clear.
We don't know whether the result has been compressed / decompressed or not.
I have been thinking about should we put the compressed status in the message header.

Reply:
Yes, it's better to add a flag.

Reply:
OK. I will add the compressed flag in another PR.

Reply:
I have to keep my modifications.
Because I use a struct to store the original message bytes, and I offer some methods to take inner structs from the struct.
After returning the inner structs, the original message bytes is still be hold by the struct. So `clone` is unavoidable.